lib/saml/user_attributes/ssoe.rb

```diff
@@ -98,25 +130,41 @@ def multifactor
       end
 
       def account_type
-        loa_current
+        result = mhv_account_type
+        result ||= safe_attr('va_eauth_dslogonassurance')
+        result ||= 'N/A'
+        result
+      end
+
+      def loa
+        { current: loa_current, highest: loa_highest }
       end
 
       def sign_in
-        if existing_user_identity?
-          existing_user_identity.sign_in
-        else
-          super
-        end
+        SAML::User::AUTHN_CONTEXTS.fetch(@authn_context)
+                                  .fetch(:sign_in)
+                                  .merge(account_type: account_type)
+      end
+
+      def to_hash
+        Hash[SERIALIZABLE_ATTRIBUTES.map { |k| [k, send(k)] }]
+      end
+
+      # Raise any fatal exceptions due to validation issues
+      def validate!
+        raise SAML::UserAttributeError, 'MHV Identifier mismatch' if mhv_id_mismatch?
```

Comment:
@patrickvinograd should we be doing this check with other ids as well.  For example, if the user is LOA3 and they logged in using ID.me credentials, their ID.me uuid should be in both the `va_eauth_uuid` and `va_eauth_gcIds` attributes.

Reply:
I think we can tighten this up over time but I don't think an ID.me mismatch is possible - any LOA3 ID.me account should be getting added to MVI. 

With MHV, we can have the case of an MHV advanced user already having a correlation in MVI, but potentially an MHV basic+ID.me LOA3 identity mapping to the same MVI record and thus having mhv_uuid (from MHV's part of the SAML payload) not match mhv_ien (from MVI). 

Could this potentially happen for DSLogon with the EDIPI parameter? Maybe :/

Reply:
(But, I think it's less of an issue for DSLogon because at this point we should only be getting DSLogon Level 2 authentications. This is really an artifact of the continuance of MHV Basic/Advanced accounts.)